### PR TITLE
Support route path in request event object

### PR DESCRIPTION
**Pull request by krambox**
I provide a pull request to extend the request event object with the root path.

With this path information, it is possible to provide a route specific event handling and interpreation. We use this information, to extend logging and monitoring informations (with good-bunyan and influx) with the hapi route.

### CODE REVIEW

**Comment by arb** [comment]
You already have the `path`. Why do you need `route` too? 

Also, you can pass in options now at the route level that will be attached to object payloads: see https://github.com/hapijs/good/blob/master/test/monitor.js#L876-L928

**Comment by krambox** [comment]
The `path` is not sufficient for us. We make a trill down analysis in Kibana / Grafana based on the service - method - route triple.
The options extension is complicated for us because we need to extend all routes in our various HapiJS micro services. With this patch every good extension can use this information. Currently we use good-bunyan with a custom response formatter function.
There may be still other opportunities to extend the event, without patching to good module?

**Comment by arb** [comment]
In your existing setup, what would the difference be between `path` and `route`? What are the differing values?

**Comment by krambox** [comment]
Typically, the routes have in our scenario parameters such as

```
server.route ({
     method: 'GET',
     path: '/hello/{user}',
...
});
```
The good response event would contain for a `GET http://localhost/hello/Adam` `path: '/hello/Adam'` and as route value  `route: '/hello/{user}'` (request.route.path)

**Comment by arb** [comment]
Do you need this data to do some kind of filter or are you actually trying to log this information somewhere?

**Comment by krambox** [comment]
I need the root.path to log and analyze in Kibana.

**Comment by arb** [comment]
While I still think you can do this with some regular expressions in your analytics tool, I don't see any real harm in adding another field. 

Please rebase this into a single commit.

**Comment by krambox** [comment]
Ok,i rebased this into one commit.

File: test/utils.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ describe('utils', () => {
             },
             method: 'POST',
             path: '/',
+            route: '/',
```
Comment (arb) [inline test/utils.js]: This isn't right. The `route` object of a request should be more than just a string according to https://github.com/hapijs/hapi/blob/master/API.md#route-public-interface

Comment (arb) [inline test/utils.js]: This isn't right. The `route` object of a request should be more than just a string according to https://github.com/hapijs/hapi/blob/master/API.md#route-public-interface

**Comment by krambox** [comment]
I adapt the test/utils.js test and extend an test in test/monitor.js. With the help from a friendly colleague, we reduced the commit mess to an single commit.

**Pull request merged**